tests/services/loading.service.spec.ts

```diff
@@ -0,0 +1,227 @@
+  let mockComponentInjector: jasmine.SpyObj<Injector>;
+  let elementRef: ElementRef;
+  let mockComponentRef: ComponentRef<MatSpinner>;
+  let mockInjector: jasmine.SpyObj<Injector>;
+  let loadingService: LoadingService;
+  let mockComponentFactory: jasmine.SpyObj<ComponentFactory<MatSpinner>>;
+  let mockComponentFactoryResolver: jasmine.SpyObj<ComponentFactoryResolver>;
+  let mockViewContainerRef: MockViewContainerRef;
+  let document: Document;
+
+  beforeEach(async () => {
+    mockMatSpinnerInstance = new MockMatSpinner();
+
+    mockRenderer = jasmine.createSpyObj<Renderer2>('Renderer2', ['createElement', 'appendChild', 'setStyle', 'addClass']);
+    mockRenderer.addClass.and.callFake((el: any, className: string) => {
+      // Simulate adding the class by appending it to the element's classList
+      el.classList.add(className);
+    });
+
+    mockComponentInjector = jasmine.createSpyObj('Injector', {
+      get: mockRenderer
+    });
+
+    elementRef = {
+      nativeElement: {
+        classList: {
+          add: jasmine.createSpy('classList.add'),
+          remove: jasmine.createSpy('classList.remove'),
+          contains: jasmine.createSpy('classList.contains'),
+          toggle: jasmine.createSpy('classList.toggle'),
+          length: 0,
+          item: jasmine.createSpy('classList.item'),
+          toString: jasmine.createSpy('classList.toString'),
+          entries: jasmine.createSpy('classList.entries'),
+          forEach: jasmine.createSpy('classList.forEach'),
+          keys: jasmine.createSpy('classList.keys'),
+          values: jasmine.createSpy('classList.values')
+        }
+      }
+    };
+
+    mockComponentRef = ({
+      location: elementRef,
+      changeDetectorRef: jasmine.createSpyObj('ChangeDetectorRef', ['detectChanges']),
+      hostView: null as any,
+      instance: mockMatSpinnerInstance,
+      injector: mockComponentInjector,
+      onDestroy: null as any,
+      destroy: () => {},
+      componentType: MatSpinner
+    } as unknown) as ComponentRef<MatSpinner>;
+
+    mockComponentFactory = jasmine.createSpyObj<ComponentFactory<MatSpinner>>({
+      create: mockComponentRef
+    });
+
+    mockComponentFactoryResolver = jasmine.createSpyObj<ComponentFactoryResolver>({
+      resolveComponentFactory: mockComponentFactory
+    });
+    mockInjector = jasmine.createSpyObj('Injector', {
+      get: mockRenderer
+    });
+
+    mockViewContainerRef = new MockViewContainerRef(mockInjector);
+
+    await TestBed.configureTestingModule({
+      providers: [
+        { provide: ComponentFactoryResolver, useValue: mockComponentFactoryResolver },
+        { provide: Injector, useValue: mockInjector },
+        { provide: MatSpinner, useClass: MockMatSpinner },
+        LoadingService
+      ]
+    }).compileComponents();
+
+    loadingService = TestBed.inject(LoadingService);
+    document = TestBed.inject(DOCUMENT);
+  });
+
+  it('should create', () => {
```

Comment:
One last nitpick, to improve readability, let's group the tests into function names & test the functionality of each function in these describe blocks. 
```githubevent.service.spec.ts``` is a good guide.

Reply:
Just refactored the tests into function names and grouped them!